### PR TITLE
feat!: update all dependencies to latest supported versions

### PR DESCRIPTION
**Pull request by shavounet**
## Why?

much cleanup, such nice

## How?

* dropped unsupported Symfony versions
* dropped unsupported PHP versions
* dropped Guzzle 6 as [it doesn't support PHP8](https://github.com/guzzle/guzzle#version-guidance)
* applied latest CS rules

Should be included in next major version

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,12 +13,12 @@ jobs:
       - run: composer install --prefer-dist --no-interaction
       - run: bin/php-cs-fixer fix --ansi --dry-run --using-cache=no --verbose
 
-  tests:
-    name: Tests
+  atoum-tests:
+    name: Atoum tests
```
Comment (Oliboy50) [inline .github/workflows/ci.yml:17]: much discrimination, such sadness

File: composer.json [hunk 2]
```diff
@@ -12,21 +12,21 @@
...
+    "php": ">=8.1",
     "ext-curl": "*",
-    "guzzlehttp/guzzle": "^6.3||^7.0",
-    "symfony/config" : "^4.4||^5.0||^6.0",
-    "symfony/dependency-injection": "^4.4||^5.0||^6.0",
-    "symfony/event-dispatcher": "^4.4||^5.0||^6.0",
-    "symfony/http-kernel": "^4.4||^5.0||^6.0",
-    "symfony/yaml" : "^4.4||^5.0||^6.0"
+    "guzzlehttp/guzzle": "^7",
+    "symfony/config" : "^5.4||^6.0",
```
Comment (Oliboy50) [inline composer.json:18]: ℹ️ [Symfony 4.4 is still supported by Symfony](https://symfony.com/releases/4.4) ("security fixes only") until November... but it should be fine to release this before November 👍

File: .github/workflows/ci.yml (hunk 1 above)
Comment (shavounet) [reply .github/workflows/ci.yml:17]: I'm planing to add a bit of rector & phpstan, that's why :grin:

File: composer.json (hunk 2 above)
Comment (shavounet) [reply composer.json:18]: As PHP8.0... but I guessed it was the time to make some cleanup. :man_shrugging:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (shavounet) [reply .github/workflows/ci.yml:17]: => #89

**Review by Oliboy50 — APPROVED** [review]
🤞

**Pull request merged**